Doc/library/re.rst

```diff
@@ -245,16 +246,24 @@ The special characters are:
    *cannot* be retrieved after performing a match or referenced later in the
    pattern.
 
-``(?imsx-imsx:...)``
-   (Zero or more letters from the set ``'i'``, ``'m'``, ``'s'``, ``'x'``,
-   optionally followed by ``'-'`` followed by one or more letters from the
-   same set.)  The letters set or removes the corresponding flags:
-   :const:`re.I` (ignore case), :const:`re.M` (multi-line), :const:`re.S`
-   (dot matches all), and :const:`re.X` (verbose), for the part of the
-   expression.  (The flags are described in :ref:`contents-of-module-re`.)
+``(?aiLmsux-imsx:...)``
+   (Zero or more letters from the set ``'a'``, ``'i'``, ``'L'``, ``'m'``,
+   ``'s'``, ``'u'``, ``'x'``, optionally followed by ``'-'`` followed by
+   one or more letters from the ``'i'``, ``'m'``, ``'s'``, ``'x'``.)
+   The letters set or removes the corresponding flags:
+   :const:`re.A` (ASCII-only matching), :const:`re.I` (ignore case),
+   :const:`re.L` (locale dependent), :const:`re.M` (multi-line),
+   :const:`re.S` (dot matches all), :const:`re.U` (Unicode matching),
+   and :const:`re.X` (verbose), for the part of the expression.
+   (The flags are described in :ref:`contents-of-module-re`.)
+   The letters ``'a'``, ``'L'`` and ``'u'`` can't be combined or follow
+   ``'-'``.  Instead using the one of them temporary removes other flags.
```

Comment:
The last sentence doesn't parse quite right.  How about "Using one of these flags temporarily removes other flags." ...? 

Also, what flags are "temporarily removed" and what does it mean to temporarily remove a flag?

I guess I'm looking for more rationale as to why some flags can appear after the dash but others can't.

Reply:
> The last sentence doesn't parse quite right. How about "Using one of these flags temporarily removes other flags." ...?

LGTM.

> Also, what flags are "temporarily removed" and what does it mean to temporarily remove a flag?

I meant the flags ASCII, LOCAL and UNICODE specified by letters 'a', 'L' and 'u'. Every of these flags affects the meaning of `\w` and case-insensitive matching. They are mutually exclusive. The UNICODE and LOCALE flags not affect the part inside `(?a:...)`. And you can nest different flags: `(?a: ascii matching (?u: unicode matching ))`.

> I guess I'm looking for more rationale as to why some flags can appear after the dash but others can't.

Because the one, and only one of these flags should be set for unambiguity. If you remove the flag that is set, you should add other flag. Thus you would need to write `(?a-u:...)` for switching from Unicode matching to ASCII matching in a string pattern. This looks cumbersome.

Reply:
Thanks for the details!  I agree with the functionality.  What about this for documentation:

"The letters ``'a'``, ``'L'`` and ``'u'`` are mutually exclusive when used as inline flags, so they can't be combined or follow ``'-'``.  Instead, when one of them appears in an inline group, it overrides any of the other two letters in the enclosing group.  This override is only in effect for the narrow inline group, and the original flags are restored outside of the group."